pkg/kube/client.go

```diff
@@ -290,13 +290,33 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 	return buf.String(), nil
 }
 
-// Update reads in the current configuration and a target configuration from io.reader
+// Deprecated; use UpdateWithOptions instead
+func (c *Client) Update(namespace string, originalReader, targetReader io.Reader, force bool, recreate bool, timeout int64, shouldWait bool) error {
+	return c.UpdateWithOptions(namespace, originalReader, targetReader, UpdateOptions{
+		Force:      force,
+		Recreate:   recreate,
+		Timeout:    timeout,
+		ShouldWait: shouldWait,
+	})
+}
+
+// UpdateOptions provides options to control update behavior
+type UpdateOptions struct {
+	Force      bool
+	Recreate   bool
+	Timeout    int64
+	ShouldWait bool
+	// Allow deletion of new resources created in this update when update failed
+	CleanupOnFail bool
+}
+
+// UpdateWithOptions reads in the current configuration and a target configuration from io.reader
 // and creates resources that don't already exists, updates resources that have been modified
 // in the target configuration and deletes resources from the current configuration that are
 // not present in the target configuration.
 //
 // Namespace will set the namespaces.
-func (c *Client) Update(namespace string, originalReader, targetReader io.Reader, force bool, recreate bool, timeout int64, shouldWait bool) error {
+func (c *Client) UpdateWithOptions(namespace string, originalReader, targetReader io.Reader, opts UpdateOptions) error {
```

Comment:
This reads much cleaner and avoids breaking API changes when we need to add other things. Thanks for the great refactor!